### PR TITLE
Add ability to open a synced realm without sync configuration.

### PR DESCRIPTION
**Pull request by leemaguire**
Add ability to open a synced realm without a flexible sync configuration.
```cpp
auto local_config = realm::db_config();
local_config.set_path(path);
local_config.enable_forced_sync_history(); // allows a synced realm to be used with a local config.
```

### CODE REVIEW

File: tests/sync/flexible_sync_tests.cpp [hunk 1]
```diff
@@ -102,6 +102,81 @@ TEST_CASE("flexible_sync", "[sync]") {
...
+            local_config.enable_forced_sync_history();
+            auto local_realm = realm::open<AllTypesObject, AllTypesObjectLink, AllTypesObjectEmbedded>(local_config);
+            auto s = local_realm.objects<AllTypesObject>().size();
+            CHECK(s == 1);
+            local_realm.write([&local_realm]() {
+                AllTypesObject o;
+                o._id = 2;
+                o.str_col = "foo";
+                local_realm.add(std::move(o));
+            });
```
Comment (dianaafanador3) [inline tests/sync/flexible_sync_tests.cpp:161]: Can we check if the local realm has the new object?

File: tests/sync/flexible_sync_tests.cpp [hunk 2]
```diff
@@ -102,6 +102,81 @@ TEST_CASE("flexible_sync", "[sync]") {
...
+                        subs.add<AllTypesObject>("foo-strings", [](auto &obj) {
+                            return obj.str_col == "foo";
+                        });
+                        subs.add<AllTypesObjectLink>("foo-link");
+                    })
+                    .get();
+            CHECK(update_success == true);
+            CHECK(synced_realm.subscriptions().size() == 2);
+
+            synced_realm.get_sync_session()->wait_for_upload_completion().get();
```
Comment (dianaafanador3) [inline tests/sync/flexible_sync_tests.cpp:131]: Do we need to sync upload and download here?, it seems like an extra step here

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,19 @@
+NEXT_RELEASE Release notes (YYYY-MM-DD)
+=============================================================
+
+### Fixed
+* None
+
+### Enhancements
+* Add `realm::db_config::enable_forced_sync_history()` which allows you to open a synced Realm
```
Comment (dianaafanador3) [inline CHANGELOG.md:8]: Can you add a little bit of context why do we need this flag?, I supposed if we are offline and cannot log the user we can make use of this, but in that case there is a user in the cache that can be used to open the sync realm offline.

File: tests/sync/flexible_sync_tests.cpp [hunk 4]
```diff
@@ -102,6 +102,81 @@ TEST_CASE("flexible_sync", "[sync]") {
...
+            auto objs = synced_realm.objects<AllTypesObject>();
+            CHECK(objs.size() == 1);
+        }
+
+        {
+            auto local_config = realm::db_config();
+            local_config.set_path(path);
+            local_config.enable_forced_sync_history();
+            auto local_realm = realm::open<AllTypesObject, AllTypesObjectLink, AllTypesObjectEmbedded>(local_config);
+            auto s = local_realm.objects<AllTypesObject>().size();
```
[the remark this answers is not in the document]
Comment (leemaguire) [reply tests/sync/flexible_sync_tests.cpp:154]: The object will be there initially, but once the sync upload commences the write will be reverted.

File: CHANGELOG.md (hunk 3 above)
Comment (leemaguire) [reply CHANGELOG.md:8]: The purpose is for when are logged out due to the refresh token expiration, and due to connectivity issues you cannot log back in.

Comment (dianaafanador3) [reply CHANGELOG.md:8]: Is this available in any other SDKs?, or Is this a request for a specific customer?

Comment (leemaguire) [reply CHANGELOG.md:8]: I didn't see it in the Swift SDK, this is for a specific customer requirement.

File: tests/sync/flexible_sync_tests.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (dianaafanador3) [reply tests/sync/flexible_sync_tests.cpp:154]: So probably we need to add this to the documentation, I supposed the use case is to open this with sync configuration when connection is restored

[the remark this answers is not in the document]
Comment (leemaguire) [reply tests/sync/flexible_sync_tests.cpp:154]: yeah, I'll add a comment for it.

**Review by dianaafanador3 — APPROVED** [review]
LGTM! Just a few comments and some minor questions

**Comment by adamski** [comment]
@leemaguire I assume in order to move the local realm to be synced, one would need to explicitly set the path to be the same in both cases, rather than use the default path given when setting it up via an app and user?

**Pull request merged**